### DISCUSSION TITLE
Yaml format for a-rtegen

### DISCUSSION THREAD

**Discussion by decoski** [toppers/users-forum#104] [Q&A]
Hello,
I found that the input of our a-rtegen tool is a yaml file, so whether there are any standard documents or tools for the preparation of this file to adapt to a-rtegen, please provide the documents or tools in this a-rtegen , thank you very much！

**Comment by mitsut** [comment]
> I found that the input of our a-rtegen tool is a yaml file,

That is slightly different.
The input file for a-rtegen is an ARXML file, which conforms to the AUTOSAR specification.(but AUTOSAR Schema version is old...)

We did not have an ECU Configuration tool at the time of ATK2 development.
And because it is difficult for manuals to create ARXML files (as they should be created by tools), we created a tool (ABREX) to convert YAML to ARXML.

You can find the ABREX manual in the utils folder of ATK2. (Sorry, only in Japanese).
As a prerequisite for writing YAML, it is necessary to know the structure of the ECU Configuration, which is also the case for ARXML files.

**Reply by decoski** [reply]
It's been a while now，Thanks.
_Os:
  Os:
    MainApp:
      DefinitionRef: OsAppMode
    OsOS:
      OsStackMonitoring: 1
      OsStatus: EXTENDED
      OsUseGetServiceId: 1
      OsUseParameterAccess: 1
      OsScalabilityClass: SC1_
May I ask if there is a document explaining the order and meaning of these parameters （MainApp,OsStatus,...）in the yaml file? 
I cann't found them in "RTE_ SPEC-01-140 " .What format do I need to refer to?

**Comment by mitsut** [comment]
@decoski 

Yaml parameter lists as follows;
https://github.com/toppers/atk2-sc1/blob/main/utils/abrex/param_info.yaml

If you want to add a container/parameter, you should change the yaml file.
